camera_system/game_area_info.py

```diff
@@ -219,68 +224,79 @@ def get_no_rotate_direction(robot) -> List[Direction]:
-        no_rotate_list += [
-            Direction.N.value for node in GameAreaInfo.node_list
-            if node.block_id != -1 and node.coord == block_coord]
-        # 西の座標
-        block_coord = Coordinate(robot.coord.x-1, robot.coord.y)
-        # 西の座標にブロックが存在する場合、東を回頭禁止方向に追加
-        no_rotate_list += [
-            Direction.E.value for node in GameAreaInfo.node_list
-            if node.block_id != -1 and node.coord == block_coord]
-        # 北の座標
-        block_coord = Coordinate(robot.coord.x, robot.coord.y-1)
-        # 北の座標にブロックが存在する場合、南を回頭禁止方向に追加
-        no_rotate_list += [
-            Direction.S.value for node in GameAreaInfo.node_list
-            if node.block_id != -1 and node.coord == block_coord]
+        # 回頭禁止方向の方位リスト
+        no_rotate_directions = []
+
+        # 走行体の東のノード
+        east_coord = Coordinate(robot.coord.x+1, robot.coord.y)
+        node_id = int(east_coord.y * 7 + east_coord.x)
+        if 0 <= node_id < 49:
+            east_node = GameAreaInfo.node_list[node_id]
+            # ノードにブロックが存在する場合、西を回頭禁止方向に追加
+            if east_node.block_id != -1:
+                no_rotate_directions += [Direction.W]
+
+        # 走行体の南のノード
+        south_coord = Coordinate(robot.coord.x, robot.coord.y+1)
+        node_id = int(south_coord.y * 7 + south_coord.x)
+        if 0 <= node_id < 49:
+            south_node = GameAreaInfo.node_list[node_id]
+            # ノードにブロックが存在する場合、北を回頭禁止方向に追加
+            if south_node.block_id != -1:
+                no_rotate_directions += [Direction.N]
+
+        # 走行体の西のノード
+        west_coord = Coordinate(robot.coord.x-1, robot.coord.y)
+        node_id = int(west_coord.y * 7 + west_coord.x)
+        if 0 <= node_id < 49:
+            west_node = GameAreaInfo.node_list[node_id]
+            # ノードにブロックが存在する場合、東を回頭禁止方向に追加
+            if west_node.block_id != -1:
+                no_rotate_directions += [Direction.E]
+
+        # 走行体の北のノード
+        north_coord = Coordinate(robot.coord.x, robot.coord.y-1)
+        node_id = int(north_coord.y * 7 + north_coord.x)
+        if 0 <= node_id < 49:
+            north_node = GameAreaInfo.node_list[node_id]
+            # ノードにブロックが存在する場合、南を回頭禁止方向に追加
+            if north_node.block_id != -1:
+                no_rotate_directions += [Direction.S]
 
         # 回頭禁止方向が2つ以上ある場合、それらの間にある方位も回頭禁止方向
-        if len(no_rotate_list) >= 2:
-            if min(no_rotate_list) < direct and direct < max(no_rotate_list):
-                for i in range(min(no_rotate_list)):
-                    no_rotate_list.append(i)
-                for i in range(max(no_rotate_list)+1, 8):
-                    no_rotate_list.append(i)
+        if len(no_rotate_directions) >= 2:
+            min_direction_value = min([direction.value for direction in no_rotate_directions])
+            max_direction_value = max([direction.value for direction in no_rotate_directions])
+            if min_direction_value < robot.direct.value < max_direction_value:
```

Comment:
```suggestion
            if min_direction_value < robot.direct.value and robot.direct.value < max_direction_value:
```
python は and で分けないといけなかったと思います。